engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailProcessor.java

```diff
@@ -0,0 +1,196 @@
+        final String orgId = event.getOrgId();
+
+        // Check if the aggregation should be persisted or not.
+        if (this.templateRepository.isEmailAggregationSupported(bundleName, applicationName, NON_INSTANT_SUBSCRIPTION_TYPES)) {
+            final EmailAggregation aggregation = new EmailAggregation();
+            aggregation.setApplicationName(applicationName);
+            aggregation.setBundleName(bundleName);
+            aggregation.setOrgId(orgId);
+
+            // Transform the event to
+            final JsonObject transformedEvent = this.baseTransformer.toJsonObject(event);
+            aggregation.setPayload(transformedEvent);
+
+            try {
+                this.emailAggregationRepository.addEmailAggregation(aggregation);
+            } catch (Exception e) {
+                // ConstraintViolationException may be thrown here and it must not interrupt the email that is being sent.
+                Log.warn("Email aggregation persisting failed", e);
+            }
+        }
+
+        // Fetch the template that will be used to hydrate it with the data.
+        final Optional<InstantEmailTemplate> instantEmailTemplate = this.templateRepository.findInstantEmailTemplate(event.getEventType().getId());
+        if (instantEmailTemplate.isEmpty()) {
+            // TODO don't even log
+            Log.errorf("there is no email template for this one :(");
+            return;
+        }
+
+        // Get the set of UUIDs that should receive an email notification for
+        // the given event.
+        final Set<String> subscribersUUIDList = this.getSubscribers(event);
+        if (subscribersUUIDList.isEmpty()) {
+            // TODO don't even log
+            Log.errorf("the subscribers' UUID list is empty :/");
+            return;
+        }
+
+        final UUID historyUUID = UUID.randomUUID();
+        final Endpoint endpoint = this.endpointRepository.getOrCreateDefaultSystemSubscription(accountId, orgId, EndpointType.EMAIL_SUBSCRIPTION);
+
+        final NotificationHistory history = NotificationHistory.getHistoryStub(endpoint, event, 0L, historyUUID);
+        history.setStatus(NotificationStatus.PROCESSING);
+        this.persistNotificationHistory(history);
+
+        // Prepare all the data to be sent to the connector.
+        final EmailConnector emailConnector = new EmailConnector(
+            endpoints,
+            subscribersUUIDList,
+            endpoint,
+            this.extractRecipientSettings(endpoints, event),
+            this.baseTransformer.toJsonObject(event).encode(),
+            instantEmailTemplate.get().getBodyTemplate(),
+            instantEmailTemplate.get().getSubjectTemplate(),
+            EmailSubscriptionType.INSTANT.isOptIn()
+        );
+
+        final JsonObject payload = JsonObject.mapFrom(emailConnector);
+
+        try {
+            this.connectorSender.send(payload, historyUUID, EndpointType.EMAIL_SUBSCRIPTION.name().toLowerCase());
```

Comment:
Things will get a bit simpler around here after https://github.com/RedHatInsights/notifications-backend/pull/2119 is merged.